test/Dialect/FIRRTL/lower-layers.mlir

```diff
@@ -1,11 +1,353 @@
 // RUN: circt-opt -firrtl-lower-layers -split-input-file %s | FileCheck %s
 
+firrtl.circuit "Test" {
+  firrtl.module @Test() {}
+
+  firrtl.layer @A bind {
+    firrtl.layer @B bind {
+      firrtl.layer @C bind {}
+    }
+  }
+  firrtl.layer @B bind {}
+
+  firrtl.extmodule @Foo(out o : !firrtl.probe<uint<1>, @A>)
+
+  //===--------------------------------------------------------------------===//
+  // Removal of Probe Colors
+  //===--------------------------------------------------------------------===//
+
+  // CHECK-LABEL: @ColoredPorts(out %o: !firrtl.probe<uint<1>>)
+  firrtl.module @ColoredPorts(out %o: !firrtl.probe<uint<1>, @A>) {}
+
+  // CHECK-LABEL: @ExtColoredPorts(out o: !firrtl.probe<uint<1>>)
+  firrtl.extmodule @ExtColoredPorts(out o: !firrtl.probe<uint<1>, @A>)
+
+  // CHECK-LABEL: @ColoredPortsOnInstances
+  firrtl.module @ColoredPortsOnInstances() {
+    // CHECK: %foo_o = firrtl.instance foo @ColoredPorts(out o: !firrtl.probe<uint<1>>)
+   %foo_o = firrtl.instance foo @ColoredPorts(out o: !firrtl.probe<uint<1>, @A>)
+  }
+
+  // CHECK-LABEL: @ColoredThings
+  firrtl.module @ColoredThings() {
+    // CHECK: %0 = firrtl.wire : !firrtl.probe<bundle<f: uint<1>>>
+    %0 = firrtl.wire : !firrtl.probe<bundle<f: uint<1>>, @A>
+    // CHECK: %1 = firrtl.ref.sub %0[0] : !firrtl.probe<bundle<f: uint<1>>>
+    %1 = firrtl.ref.sub %0[0] : !firrtl.probe<bundle<f: uint<1>>, @A>
+    // CHECK-NOT: firrtl.cast
+    %2 = firrtl.ref.cast %1 : (!firrtl.probe<uint<1>, @A>) -> !firrtl.probe<uint<1>, @A::@B>
+  }
```

Comment:
Including a check of removal of a layer-colored type under a `when` block would be good.